credentialdigger/cli/get_discoveries.py

```diff
@@ -0,0 +1,279 @@
+
+
+def print_discoveries(discoveries, repo_url):
+    """ Print discoveries on the terminal in a tabular format.
+
+    Parameters
+    ----------
+        discoveries: list
+            List of the discoveries to be printed.
+        repo_url : str
+            The url of the repo from which we retrieved the discoveries
+    """
+    with console.status(f'[bold]Processing {len(discoveries)} discoveries...'):
+        discoveries_df = pd.DataFrame(discoveries)
+        # Remove the repo_url column since it has been passed as an argument
+        del discoveries_df['repo_url']
+        # Remove timestamp and rule_id columns because they are not
+        # quite relevant
+        del discoveries_df['timestamp']
+        del discoveries_df['rule_id']
+
+        # Convert `int` columns to `str` to be eventually rendered.
+        discoveries_df['id'] = discoveries_df['id'].astype(str)
+        discoveries_df['line_number'] = discoveries_df['line_number'].astype(
+            str)
+
+        # Convert to list and insert column names
+        discoveries_list = discoveries_df.values.tolist()
+        columns = ['id', 'file_name', 'commit_id', 'line_number',
+                   'snippet', 'state']
+
+        table = Table(
+            title=f'Discoveries found in "{repo_url}"',
+            pad_edge=False, show_lines=True)
+
+        for c in columns:
+            table.add_column(c)
+        for row in discoveries_list:
+            table.add_row(*row)
+
+        console.print(table)
+
+
+def discoveries_to_csv(discoveries):
+    """ Generate CSV from list of discoveries
+
+    Parameters
+    ----------
+        discoveries: list
+            List of discoveries from which to generate the CSV
+
+    Returns
+    -------
+        str
+            A string containing CSV obtaining from the original list of
+            discoveries.
+    """
+    try:
+        stringIO = io.StringIO()
+        csv_writer = csv.DictWriter(stringIO, discoveries[0].keys())
+        csv_writer.writeheader()
+        csv_writer.writerows(discoveries)
+        csv_data = stringIO.getvalue()
+    except IndexError as error:
+        logger.error(error)
+    except Exception as exception:
+        logger.exception(exception)
+
+    return csv_data
+
+
+def export_csv(discoveries, client, save=False):
+    """ Export discoveries as a CSV file
+
+    Parameters
+    ----------
+        discoveries: list
+            List of discoveries from which to generate the CSV
+        client: `credentialdigger.Client`
+            Instance of the client from which we retrieve rules
+        save: bool
+            If True, we do not ask the user to enter a file path for the CSV
+            to be exported
+    """
+    # Check if --save is specified
+    if save is False:
+        path = ''
+        # Read the export path from the console's input
+        while path == '':
+            path = console.input('Path to export CSV:')
```

Comment:
if `save` is false, then why shall we specify a path? If save is false, then no path is needed

Reply:
We ask the user to provide the CLI with a path if the number of discoveries is 30 and they enter 'Y' (to export the discoveries).

If --save is specified (not False), we won't show the message requesting the user to specify the path and we will just export the discoveries directly.

If we happen to access export_csv with save == False, then this means that the repo has more than 30 discoveries and the user wants to export them instead of printing them on the Terminal.